src/Mutation/NodeTraverserFactory.php

```diff
@@ -49,17 +50,17 @@ final class NodeTraverserFactory
     /**
      * @param NodeVisitor[] $extraVisitors
      */
-    public function create(array $extraVisitors): PriorityNodeTraverser
+    public function create(array $extraVisitors): PrioritizedVisitorsNodeTraverser
```

Comment:
The choice here is to make `PrioritizedVisitorsNodeTraverser` a full-pledge adapter and no longer depend on `NodeTraverserInterface` when consuming the traverser. It is not a final decision but there is no need to be more generic for now and facilitates a few things, since we can be sure about the traverser downstream when using it.